worker/transfer.go

```diff
@@ -442,12 +453,9 @@ func parallelDownloadSlab(ctx context.Context, sp storeProvider, ss object.SlabS
 			// make sure slow hosts are not not used for consecutive downloads
 			if errors.Is(resp.err, errDownloadSectorTimeout) {
 				timings[resp.hostIndex] = int64(resp.dur) * 10
-			}
-
-			// try next host
-			hostIndex := workerForShard(resp.req.shardIndex)
-			if hostIndex != -1 {
-				launchWorker(resp.req)
+				if overdrive < maxOverdrive {
+					overdrive++ // add more overdrive
```

Comment:
I think there might be a misunderstanding here due to the name. `overdrive` is not actually the current number of overdrive workers. As you might have noticed we are not decrementing it ever. Instead it is the soft limit for adding overdrive while `maxOverdrive` is the hard limit. That's why it's not close to launching workers because it's unrelated to how many workers we have launched already.